toucan_connectors/redshift/redshift_database_connector.py

```diff
@@ -347,39 +367,20 @@ def describe(self, data_source: RedshiftDataSource) -> dict[str, Any]:
             for col in res
         }
 
-    def _table_infos_rows(self) -> list[tuple[str, str, str, str, str]]:
-        """Get rows of (database, schema, table name, column name, column type)"""
-        with self._get_cursor(database=self.default_database) as cursor:
-            res = cursor.execute(
-                """
-                SELECT
-                    table_catalog,
-                    table_schema,
-                    table_name,
-                    column_name,
-                    data_type
-                FROM
-                    information_schema.columns
-                WHERE
-                    table_schema NOT IN ('information_schema', 'pg_catalog')
-                ORDER BY
-                    table_schema,
-                    table_name,
-                    ordinal_position;
-                """
+    def _db_table_info_rows(self, database: str) -> list[tuple[str, str, str, str]]:
+        with self._get_cursor(database) as cursor:
+            """Get rows of (schema, table name, column name, column type)"""
+            cursor.execute(
+                r"""SELECT "schemaname", "tablename", "column", "type" FROM PG_TABLE_DEF WHERE schemaname = 'public';"""
```

Comment:
Could fail if the db user is not allowed to query this table. 
But it's okay to try IMO.